appserver/tests/rq.sh

```diff
@@ -39,145 +39,13 @@
-      -e s@'<parameter>'@@g -e s@'</parameter>'@@g \
-      -e s@'<name>'@@g -e s@'</name>'@@g \
-      -e s@'<value>'@'='@g -e s@'</value>'@' '@g -e s@'<value/>'@' '@g 
-
-
-  if [ ${PIPESTATUS[0]} -ne 0 ] ; then
-    # exit with curl status code
-    return ${PIPESTATUS[0]}
-  fi
-}
-
-#
-# Returns true if the test job was triggered by the current PARENT_ID and given TEST_ID
-#
-# Args: params array to match the job
-#
-is_test_job_match(){
-  local array=${2}
-  local match_params=(${array[*]})
-  set +e
-  local job_param=(`get_test_job_params ${1}`)
-  local error_code=${?}
-  set -e
-
-  if [ ${error_code} -ne 0 ] ; then
-    # no match
-    echo false
-    return 0
-  fi
-  # match provided params with job params
-  local i=0
-  while [ ${i} -lt ${#match_params[*]} ]
-  do
-    local match_entry=(`key_val_as_list  ${match_params[${i}]}`)
-    job_value=`get_value_from_key_val_array ${match_entry[0]} "${job_param[*]}"`
-    if [ "${job_value}" != "${match_entry[1]}" ] ; then
-      echo false
-      return 0
-    fi
-    i=$((i+1))
-  done
-  # match
-  echo true
-}
-
-#
-# Gets the last build number of the test job.
-# This will include the currently on-going runs.
-#
-get_test_job_last_build_number(){
-  local url="${GLASSFISH_REMOTE_QUEUE_URL}/api/xml?xpath=//lastBuild/number/text()"
-  curl "${url}"
-  local error_code=${?}
-  if [ ${error_code} -ne 0 ] ; then
-    exit 1
-  fi
-}
-
-#
-# Find a test job for TEST_ID
-#
-# Args: TEST_ID PREVIOUS_LAST_BUILD
-#
-find_test_job(){
-    local previous_last_build=${1}
-    local last_build=`get_test_job_last_build_number`
-
-    # nothing running and nothing new completed
-    if [ "${previous_last_build}" = "${last_build}" ] ; then
-      return 1
-    fi
-
-    # look into the newly completed run
-    local i=$((previous_last_build+1))
-    while [ ${i} -le ${last_build} ]
-    do
-      local params
-      params[0]="BRANCH=${branch}"
-      params[1]="TEST_IDS=${test_ids_triggerd}"
-      params[2]="PR_NUMBER"
-      params[3]="FORK_ORIGIN=${fork_origin}"
-      if `is_test_job_match ${i} "${params[*]}"` ; then
-        # the triggered run is already completed
-        echo ${i}
-        return 0
-      fi
-      i=$((i+1))
-    done
-
-    # not found
-    return 1
-}
-
 USAGE="Usage:\n\n 1. rq.sh -l ---> List all available test identifiers without running them\n\
-	   2. rq.sh -b <branch> -a ---> For running all tests\n\
+	   2. rq.sh -b <branch> -a ---> For running all tests in remote branch\n\
 	   3. rq.sh -b <branch> -g <test_group_name> ---> For running a test group\n\
-	   4. rq.sh -b <branch> -t \"<test_id1> <test_id2> <test_id3>\" ---> For running a space separated list of tests"
+	   4. rq.sh -b <branch> -t \"<test_id1> <test_id2> <test_id3>\" ---> For running a space separated list of tests\n\
+	   5. rq.sh -u <glassfish binary url> -a ---> For running all tests with GlassFish binary provided in the url.-u option works with -a, -g and -t options as well\n\
```

Comment:
Maybe this can be changed to reflect the combinations
rq.sh -u <glassfish binary url> -a | -u |-t
Also need to  mention that this should be http url only.